packages/lib/src/storybook/storybookArgTypes.ts

```diff
@@ -21,61 +21,97 @@ export interface StoryArgType extends InputType {
       };
 }
 
-export const storybookArgTypes: Record<string, StoryArgType> = {
-  baseFontSize: {
-    description:
-      'The base font size passed to the LeafyGreenProvider that wraps the component',
-    control: { type: 'radio' },
-    options: [14, 16],
-  },
-  updatedBaseFontSize: {
-    description:
-      'The base font size passed to the LeafyGreenProvider that wraps the component. Uses the updated font size values for Euclid Circular A.',
-    control: { type: 'radio' },
-    options: [13, 16],
-  },
-  darkMode: {
-    description: 'Render the component in dark mode.',
-    control: 'boolean',
-  },
-  children: {
-    description: 'Element rendered inside the component',
-    control: 'text',
-  },
-  as: {
-    description:
-      'The component will be rendered in HTML as the element selected here',
-    options: IntrinsicElements,
-    type: { name: 'string' },
-    control: { type: 'select' },
-    defaultValue: 'button',
-  },
+type BaseArgTypes = 'boolean' | 'number' | 'range' | 'text' | 'color' | 'date';
+type ArgTypeWithOptions = 'radio' | 'check' | 'select' | 'multi-select';
+
+function argType(
+  type: BaseArgTypes,
+): (description?: string, config?: Record<string, any>) => StoryArgType;
+function argType(
+  type: ArgTypeWithOptions,
+): (
+  options: Array<any>,
+  description?: string,
+  config?: Record<string, any>,
+) => StoryArgType;
+function argType(
+  type: BaseArgTypes | ArgTypeWithOptions,
+): (...args: any) => StoryArgType {
+  if (['radio', 'check', 'select', 'multi-select'].includes(type)) {
+    return (
+      options: Array<any>,
+      description?: string,
+      config?: Record<string, any>,
+    ) =>
+      ({
+        description,
+        options,
+        control: { type, ...config },
+      } as StoryArgType);
+  } else
+    return (description?: string, config?: Record<string, any>) =>
+      ({
+        description,
+        control: { type, ...config },
+      } as StoryArgType);
+}
+
+export const StorybookArgTypes = {
+  Boolean: argType('boolean'),
+  Number: argType('number'),
+  Range: argType('range'),
+  Select: argType('select'),
+  Radio: argType('radio'),
+  Check: argType('check'),
+  Text: argType('text'),
+  None: { control: 'none' },
+} as const;
+
+/**
+ * Default configurations for storybook arg types
+ */
+export const defaultStorybookArgTypes: Record<string, StoryArgType> = {
+  baseFontSize: StorybookArgTypes.Radio(
+    [14, 16],
+    'The base font size passed to the LeafyGreenProvider that wraps the component',
+  ),
+  updatedBaseFontSize: StorybookArgTypes.Radio(
+    [13, 16],
+    'The base font size passed to the LeafyGreenProvider that wraps the component. Uses the updated font size values for Euclid Circular A.',
+  ),
+  darkMode: StorybookArgTypes.Boolean('Render the component in dark mode.'),
+  children: StorybookArgTypes.Text('Element rendered inside the component'),
+  as: StorybookArgTypes.Select(
+    IntrinsicElements,
```

Comment:
we probably want to be a little bit more restrictive here, right now `br` is breaking IconButton, for example